File: week6/pset6/sentimental-readability/readability.py
# TODO: readability

from cs50 import get_string


def main():
    # Getting input from user
    text = get_string("Text: ")

    # Assigning counting
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # calculating per 100 words
    L = letters / words * 100
    S = sentences / words * 100

    # Coleman-Liau index formula
    index = 0.0588 * L - 0.296 * S - 15.8

    # roundoff index
    n = round(index)

    # printing results
    if n > 1 and n < 16:
        print(f"Grade {n}")
    elif n <= 1:
        print("Before Grade 1")
    elif n >= 16:
        print("Grade 16+")

# counting letters


def count_letters(text):
    count = 0
    for i in range(len(text)):
        if (text[i].isalpha()):
            count += 1
    return count

# counting words


def count_words(text):
    count = 1
    for i in range(len(text)):
        if (text[i] == ' '):
            count += 1
    return count

# counting sentences


def count_sentences(text):
    count = 0
    for i in range(len(text)):
        if (text[i] == '.') or (text[i] == '!') or (text[i] == '?'):
            count += 1
    return count


# executing main
if __name__ == "__main__":
    main()

File: week2/pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);

int main(void)
{
    // Getting input from user
    string text = get_string("Text: ");

    // Assigning counting
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    // calculating per 100 words
    float L = letters / words * 100;
    float S = sentences / words * 100;

    // Coleman-Liau index formula
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // roundoff index
    int n = round(index);

    // printing results
    if (n > 1 && n < 16)
    {
        printf("Grade %i\n", n);
    }
    else if (n <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (n >= 16)
    {
        printf("Grade 16+\n");
    }
}

// counting letters
int count_letters(string s)
{
    int counter = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]))
        {
            counter++;
        }
    }
    return counter;
}

// counting words
int count_words(string s)
{
    int counter = 1;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == ' ')
        {
            counter++;
        }
    }
    return counter;
}

// counting sentences
int count_sentences(string s)
{
    int counter = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if ((s[i] == '.') || (s[i] == '!') || (s[i] == '?'))
        {
            counter++;
        }
    }
    return counter;
}